### DISCUSSION TITLE
Are we hitting the limits of the current model ?

### DISCUSSION THREAD

**Discussion by Christian-G-UT** [neherlab/covid19_scenarios#528] [General]
Looking at regions that are quite far in the epidemic (like Lombardy in Italy) it does not seem possible to fit the curves of the current model simultaneously to the number of hospitalized, ICU and deaths data. All these three figures appear quite reliable so it should be possible to fit them with the right model, but I couldn't succeed. Lombardy has passed the peak but is currently on a plateau which I was not able to correctly model with the current implementation. Yet modeling the end of the epidemic is as important as modeling the beginning. 
I would first like to have the opinion of the developers, on whether they agree that the current hospitalized, ICU and deaths data in Lombardy is not possible to be appropriately fitted. 
My feeling is that the model has some fixed clinical dynamics parameters which could benefit from tuning. The model available here https://gabgoh.github.io/COVID/index.html , while generally much less complex, has for example four parameters that can possibly improve the model representativness:
- Time to hospitalization
- Length of hospital stay
- Time from end of incubation to death.

Of course each of these parameters is heavily dependent on the country and the phase of the epidemic in which it is but I think there starts to be enough information available to have reasonable estimates for each. These do not have to be made as easily accessible as the main simulation parameters, they could be more hidden, similar to the age group specific parameters. There could also be preset values depending on a choice of the phase of the epidemic. 
I would appreciate having the developing team's opinion on this or on any other way to improve the fitting of the model for regions which are in the downward phase of the epidemic.
Thanks.

**Comment by nnoll** [comment]
I would disagree that the model can not fit the data. There are more than enough parameters to fit the set of curves we see. After some parameter tweaking, I arrived at [this result](https://covid19-scenarios.org/?v=1&q=~(current~'Custom~containment~(~(id~'e9627a35-a9b3-4b0c-8eb2-2e76310be547~name~'Intervention*20*231~color~'*23bf5b17~mitigationValue~65~timeRange~(tMin~1583352000000~tMax~1598918400000))~(id~'63f1e87b-d38b-4793-951b-ad2da5577a88~name~'Intervention*20*232~color~'*23dd2365~mitigationValue~40~timeRange~(tMin~1585767600000~tMax~1601492400000)))~population~(cases~'ITA-Lombardia~country~'Italy~hospitalBeds~34473~ICUBeds~1067~importsPerDay~0.1~initialNumberOfCases~989~populationServed~10060574)~epidemiological~(infectiousPeriod~5~latencyTime~3.2~lengthHospitalStay~6.4~lengthICUStay~4~overflowSeverity~2~peakMonth~0~r0~3.9~seasonalForcing~0)~simulation~(simulationTimeRange~(tMin~1581364800000~tMax~1598918400000)~numberStochasticRuns~0)~ageDistribution~(0-9~4994996~10-19~5733447~20-29~6103437~30-39~6998434~40-49~9022004~50-59~9567192~60-69~7484860~70-79~6028907~80*2b~4528548))). Some more tweaking can get you even closer.

It is important to emphasize that our model has no hard-coded parameter assumptions within - all parameters are exposed to the user via our UI. For example, all 3 of your example parameters are found within the Epidemiological card. For example, the infectious period + latency time is the serial interval and defines the time to hospitalization.

Estimating the plateau depends critically on the mitigation measures: both the date and the strength/efficacy. The dates we can get/are working to get access to, however the strengths will have to be estimated directly from the data (measuring the growth rate before & after mitigation). This is on our TODO list. 

Currently we initialize the results for each scenario fit to the dynamics of the early epidemic so you are right that our presets will likely estimate the late dynamics incorrectly. This is a sign that social distancing is working, however continuing the fit deeper into the epidemic in an automated fashion is somewhat non-trivial as we need information on when each mitigation measure was put in place.

**Comment by Christian-G-UT** [comment]
Thanks for your reply. The provided tuning seems indeed sensible.
The first thing I learned from your link of the results is that we can actually have non integer parameters for some of the parameters. I later realized that when ticking Run automatically it will run the simulation with non integer parameters. However when it is run manually it complains when having non integer parameters (for latency for example). I don't know whether this is intended behavior or not, but it is at least good to know how it works.
Finally I would just like to check whether I got the relationships between parameters correctly:
time to hospitalization = infectious period + latency
time from end of incubation to death = infectious period + hospital stay + ICU stay
(or time from end of incubation to death = infectious period + ICU stay ?)
Thanks again.

**Reply by ivan-aksamentov** [reply]
@Christian-G-UT Thanks for your feedback! Regarding the non-integer parameters, this has been indeed fixed in one of the hotfixes a couple of days ago.

If you are interested in latest features, feel free to also check out our pre-prelease deployment at:
https://master.covid19-scenarios.org
(`master` git branch is automatically deployed there)

and the discussions in our pull requests:
https://github.com/neherlab/covid19_scenarios/pulls
